### PR TITLE
:tada: added ignoredRoles option

### PR DESCRIPTION
**Pull request by DedaDev**
Hi, I made a small contribution to your package, now user can add a list of ignored roles by ID or role name.

### CODE REVIEW

**Comment by MrAugu** [comment]
The implementation has not been added because you can already do it in the library.

**Comment by DedaDev** [comment]
Are you sure?

**Comment by MrAugu** [comment]
```js
const DiscordAntiSpam = require("discord-anti-spam"); const AntiSpam = new DiscordAntiSpam({
  exemptRoles: (role) => [ArrayOfIds].includes(role.id)
});```
This way.

**Comment by MrAugu** [comment]
Exempt functions are next-gen filtering, they were added in v2 so even public bots can make use of the module. There isn't too much talk going on around them because the author module declined for them to be in the example.

**Comment by Androz2091** [comment]
If we look at it this way, we can already ignore guilds via the excempt function... So why an `ignoredGuilds` opt?

Knowing that for some time now, it has been the users who decide whether or not they want the message checked (with the AntiSpam.message() method). These filter options lose some meaning, it would rather be up to the user to filter before AntiSpam. message().

**Comment by DedaDev** [comment]
`exemptRoles` was no documented, but anyways. I would merge this, it's much neat.

File: index.js [hunk 1]
```diff
@@ -62,6 +63,7 @@ class antiSpam extends Events.EventEmitter {
     if (this.ignoredGuilds.includes(message.guild.id)) return;
     if (this.ignoredUsers.includes(message.author.id)) return;
     if (this.ignoredChannels.includes(message.author.id)) return;
+    if(message.member.roles.filter(role=> this.ignoredRoles.includes(role.id) || this.ignoredRoles.includes(role.name)).size > 0) return;
```
Comment (Androz2091) [inline index.js]:
```js
if (this.ignoredRoles.some((r) => message.member.roles.has(r))) return;
```
I think it would be better this way

**Comment by MrAugu** [comment]
There are conflicting files. Fix them.
As we need to run lists against each other more times we can do:

```js
function hasComonItem (collection, array) {
  var hasComon = false;
  for (const object of collection) {
    if (hasComon === true) break;
    if (array.includes(object.id) hasComon = true;
  }
  return hasComon;
}
```
Then in library:
```js
if (hasComonItem(message.member.roles, ignoredRoles)) return;
```

**Comment by Androz2091** [comment]
`some` method works too and is simpler...

**Comment by MrAugu** [comment]
We need function both for roles and channels.

**Comment by MrAugu** [comment]
If conflicts are not going to be resolved and the Change requests fulfilled, this PR can be closed and feature should be implemented outside the PR.

**Pull request merged**